File: 012_Vedio_Read_Wri/012_Vedio_Read_Wri/012_Vedio_Read_Wri.cpp
#include<opencv2/opencv.hpp>
#include<iostream>

using namespace cv;
using namespace std;

int main(int argc, char** argv) {
	// 打开摄像头
	VideoCapture capture(0); 

	// 打开文件
	//VideoCapture capture;
	//capture.open("H:\\OpenCV_Learning\\012_Vedio_Read_Wri\\vtest.avi");
	if (!capture.isOpened()) {
		printf("could not read this video file...\n");
		return -1;
	}
	int w = static_cast<int>(capture.get(CAP_PROP_FRAME_WIDTH));
	int h = static_cast<int>(capture.get(CAP_PROP_FRAME_HEIGHT));
	/*Size S = Size((int)capture.get(CAP_PROP_FRAME_WIDTH),
		(int)capture.get(CAP_PROP_FRAME_HEIGHT));*/
	int fps = capture.get(CAP_PROP_FPS);
	int type = VideoWriter::fourcc('m', 'p', '4', 'v');
	printf("current fps : %d \n, current type : %d \n", fps, type);
	VideoWriter writer("test.mp4", type, fps, Size(w, h), true);

	Mat frame;
	namedWindow("camera-demo", WINDOW_AUTOSIZE);
	while (capture.read(frame)) {
		imshow("camera-demo", frame);
		writer.write(frame);
		char c = waitKey(50);
		if (c == 27) {
			break;
		}
	}
	capture.release();
	writer.release();
	waitKey(0);
	return 0;
}

File: Topic/012_Vedio_Read_Wri/012_Vedio_Read_Wri.py
import cv2 as cv
import numpy as np

# capture = cv.VideoCapture("H:\\OpenCV_Learning\\012_Vedio_Read_Wri\\vtest.avi")
capture = cv.VideoCapture(0)
height = capture.get(cv.CAP_PROP_FRAME_HEIGHT)
width = capture.get(cv.CAP_PROP_FRAME_WIDTH)
count = capture.get(cv.CAP_PROP_FRAME_COUNT)
fps = capture.get(cv.CAP_PROP_FPS)
type = capture.get(cv.CAP_PROP_FORMAT)
print(height, width, count, fps, type)
out = cv.VideoWriter("test.mp4", cv.VideoWriter_fourcc('D', 'I', 'V', 'X'),  # mp4编码格式，官网文档可查
                     15, (np.int(width), np.int(height)), True)
while True:
    ret, frame = capture.read()
    if ret is True:
        cv.imshow("vedio_show", frame)
        out.write(frame)
        c = cv.waitKey(50)
        if c == 27:
            break
    else:
        break

capture.release() # 释放资源
out.release()


